src/libexpr/attr-set.cc

```diff
@@ -2,51 +2,83 @@
 #include "eval-inline.hh"
 
 #include <algorithm>
-
+#include <string.h>
 
 namespace nix {
 
+std::unordered_set<Bindings::Names, Bindings::NamesHasher>       * Bindings::pnamesTable = new std::unordered_set<Bindings::Names, Bindings::NamesHasher>      (0x4000, Bindings::NamesHasher());
```

Comment:
Should we give a name to this type? Also, `auto` on the LHS at least.

Reply:
(And a bunch more places for `auto` throughout the PR)

Reply:
Yes, I just prefer to avoid ```auto``` during development to avoid surprises (```auto``` could deduce another type, not what I think of when read the code)
But it might have sense to shrink well-tested code.